Code/Editor/Settings.cpp

```diff
@@ -638,14 +638,16 @@ void SEditorSettings::Save()
     // --- Settings Registry values
 
     // Prefab System UI
-    AzFramework::ApplicationRequests::Bus::Broadcast(
-        &AzFramework::ApplicationRequests::SetPrefabSystemEnabled, prefabSystem);
+    AzFramework::ApplicationRequests::Bus::Broadcast(&AzFramework::ApplicationRequests::SetPrefabSystemEnabled, prefabSystem);
 
     AzToolsFramework::Prefab::PrefabLoaderInterface* prefabLoaderInterface =
         AZ::Interface<AzToolsFramework::Prefab::PrefabLoaderInterface>::Get();
     prefabLoaderInterface->SetSaveAllPrefabsPreference(levelSaveSettings.saveAllPrefabsPreference);
 
-    SaveSettingsRegistryFile();
+    if (!isEditorClosing)
```

Comment:
Could you clarify the intention behind this change? I'd think we would want to save settings on editor close from a user standpoint even if it might not be desirable in a testing scenario. If the goal is to prevent the prefab enabled setting from being saved when we pass it to a test, I think we should find a different solution that doesn't impact normal editor operation.

Reply:
This is something that Danilo did for a quick fix of the saving .setreg file bug: https://github.com/o3de/o3de/pull/5717
I just temporary used that fix for debugging for now.

Reply:
This is from @AMZN-daimini's fix for setreg file overridden bug: https://github.com/o3de/o3de/pull/5717 .
I just temporarily used this fix for now for debugging.

Reply:
@AMZN-daimini has a prototype(https://github.com/o3de/o3de/pull/3025/files) to track where the settings were source from, and from the prototype there's a EditorSettingsOriginTracker that could be used to determine the original settings from the editorprefrences.setreg file. That could be a long term fix of the setreg file overridden bug.